doc/modules/mod_inbox.md

```diff
@@ -27,49 +27,98 @@ Inbox currently supports the following DBs:
 * PgSQL via native driver
 * MSSQL via ODBC driver
 
-
 ### Legacy MUC support
 Inbox comes with support for the legacy MUC as well. It stores all groupchat messages sent to
 room in each sender's and recipient's inboxes and private messages. Currently it is not possible to
 configure it to store system messages like [subject](https://xmpp.org/extensions/xep-0045.html#enter-subject) 
 or [affiliation](https://xmpp.org/extensions/xep-0045.html#affil) change.
 
+### Filtering and ordering
+
+Inbox query results may be filtered by time range and sorted by timestamp.
+By default, `mod_inbox` returns all conversations, listing the ones updated most recently first.
+
+A client may specify three parameters:
+
+* Start date for the result set (variable `start`, value: ISO timestamp)
+* End date for the result set (variable `end`, value: ISO timestamp)
+* Order by timestamp (variable `order`, values: `asc`, `desc`)
+
+They are encoded inside a standard XMPP [Data Forms](https://xmpp.org/extensions/xep-0004.html) format.
+Dates must be formatted according to [XMPP Date and Time Profiles](https://xmpp.org/extensions/xep-0082.html).
+See the example below.
+
+Your client application may request the currently supported form with IQ get:
+
+```
+Client:
+
+<iq type='get' id='c94a88ddf4957128eafd08e233f4b964'>
+  <query xmlns='erlang-solutions.com:xmpp:inbox:0'/>
+</iq>
+
+Server:
+
+<iq from='alicE@localhost' to='alicE@localhost/res1' id='c94a88ddf4957128eafd08e233f4b964' type='result'>
+  <query xmlns='erlang-solutions.com:xmpp:inbox:0'>
+    <x xmlns='jabber:x:data' type='form'>
+      <field type='hidden' var='FORM_TYPE'><value>erlang-solutions.com:xmpp:inbox:0</value></field>
+      <field var='start' type='text-single'/>
+      <field var='end' type='text-single'/>
+      <field var='order' type='list-single'>
+        <value>desc</value>
+        <option label='Ascending by timestamp'><value>asc</value></option>
+        <option label='Descending by timestamp'><value>desc</value></option>
+      </field>
+    </x>
+  </query>
+</iq>
+```
+
 ### Example Request
 
 ```
 Alice sends:
 
 <message type="chat" to="bOb@localhost/res1" id=”123”>
-<body>Hello</body>
+  <body>Hello</body>
 </message>
 
 Bob receives:
 
 <message from="alicE@localhost/res1" to="bOb@localhost/res1" id=“123” xml:lang="en" type="chat">
-<body>Hello</body>
+  <body>Hello</body>
 </message>
 
 Alice sends:
 
-<iq type="get" id="10bca">
-<inbox xmlns=”erlang-solutions.com:xmpp:inbox:0” queryid="b6"/>
+<iq type="set" id="10bca">
+  <inbox xmlns=”erlang-solutions.com:xmpp:inbox:0” queryid="b6">
+    <x xmlns='jabber:x:data' type='form'>
```

Comment:
@fenek Is it possible to send and "empty" request like before? Or does client always have to provide `start`, `end` and `order`?

Reply:
Yes, it is (but the IQ type must still be "set") possible to send empty query. Good catch, I'll clarify it in the doc.

Reply:
Fixed.